docs/transfers_submitter.md

```diff
@@ -0,0 +1,89 @@
+to be used as a transfer sources. For that, it relies on multiple RSE
+attributes and on the configured protocols. This section provides a summary
+of what configuration parameters can influence the submitter at this stage.
+
+We will use the notation `section/option` to speak about a configuration
+value to be set in `rucio.cfg` like this:
+
+```text
+[section]
+option = value
+```
+
+The submitter will start by retrieving all the possible sources from the
+database. If the configuration value `conveyor/filter_transfertool` is set, it
+will avoid source RSEs which don't have the `transfertool` RSE attribute
+set to the correct value. Except for a particular case: when submitter is
+run with a list of transfertools in `conveyor/transfertool` and multi-hoping
+is enabled. In this case the `transfertool` RSE attribute will be ignored at
+this stage, because we assume a will to multi-hop between transfertools.
+
+In the following step, the submitter will skip all sources which don't
+respect the administrative constraints. For example, it will ignore source
+RSEs with `availability_read=False` (unless the submitter is run with
+`--ignore-availability`). It also respects the `restricted_read` and
+`restricted_write` RSE attributes for the source and the destination.
+
+Some request attributes will impact the source selection. For example, submitter
+will skip source RSEs which don't match the `source_replica_expression` or
+`allow_tape_source` conditions. It will also ignore requests witch require a
+`transfertool` that this submitter cannot use. The request attributes are
+either inherited from the rule, or set by another transfer daemon
+(for example: preparer)
+
+The next step is to perform the path computation. At this stage, submitter
+uses the distance between RSEs to perform shortest-path computations. If
+multi-hopping is enabled via `transfers/use_multihop`, then the configuration
+value `transfers/hop_penalty` + the RSE attributes `available_for_multihop`
+and `hop_penalty` will influence the distances for multi-hop paths.
+Each hop, even for single-hop transfers, must respect the protocol
+compatibility between the source of the hop and its destination. The
+[SCHEME_MAP](https://github.com/rucio/rucio/blob/1b8ca368523d13fd11bc0b32c14528f2fcec778b/lib/rucio/common/constants.py#L48)
+constant defines the compatibility between protocols. Only protocols with
+non-zero `third_party_copy_read` will be considered for source RSEs, ordered
+by priority. Same for the destination: `third_party_copy_write` is used.
+
+**Note**: distances between RSEs which are set by the administrator via
+
+```shell
+rucio-admin rse add-distance --distance 1 --ranking 1 RSE1 RSE2
+```
+
+Once all valid paths are found, after all the filtering done previously,
+the paths are ordered using the following simple
+[rules](https://github.com/rucio/rucio/blob/3d6a672da0401c57b5ea4dfe2037ca83a7bfbadd/lib/rucio/core/transfer.py#L921)
```

Comment:
We already discussed this, just putting my comments here as well.
I think it would be good to write down the rules in the doc directly (Still keep the link) and also explain what `source_ranking` and `distance_ranking` are. 

Secondly I think some paragraph about retries should be added too.

Otherwise GTG